Src/FluentAssertions/Equivalency/EnumerableEquivalencyValidatorExtensions.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Equivalency
+{
+    internal static class EnumerableEquivalencyValidatorExtensions
```

Comment:
Why are these extension methods if they are only used by the `EnumerableEquivalencyValidator`?

Reply:
Purely make them extensions methods on `AssertionScope`,  so `AssertCollectionsHaveSameCount` can call them in a more fluent way instead of passing `AssertionScope` as a parameter to each.